script/job_generator/generate_job_yaml.py

```diff
@@ -0,0 +1,248 @@
+"""Create GitLab-CI job description written in yaml from the job matrix."""
+
+from typing import List, Dict, Tuple
+from typeguard import typechecked
+import os, yaml
+from packaging import version as pk_version
+
+
+from alpaka_job_coverage.globals import *  # pylint: disable=wildcard-import,unused-wildcard-import
+
+
+@typechecked
+def get_env_var_name(variable_name: str) -> str:
+    """Transform string to a shape, which is allowed as environment variable.
+
+    Args:
+        variable_name (str): Variable name.
+
+    Returns:
+        str: Transformed variable name.
+    """
+    return variable_name.upper().replace("-", "_")
+
+
+@typechecked
+def job_prefix_coding(job: Dict[str, Tuple[str, str]]) -> str:
+    """Generate prefix for job name, depending of the available software versions.
+
+    Args:
+        job (Dict[str, Tuple[str, str]]): Job dict.
+
+    Returns:
+        str: Job name Prefix.
+    """
+    version_str = ""
+
+    return version_str
+
+
+@typechecked
+def job_image(job: Dict[str, Tuple[str, str]], container_version: float) -> str:
+    """Generate the image url deppending on the host and device compiler and the selected backend
+    of the job.
+
+    Args:
+        job (Dict[str, Tuple[str, str]]): Job dict.
+        container_version (float): Container version tag.
+
+    Returns:
+        str: Full container url, which can be used with docker pull.
+    """
+    container_url = "registry.hzdr.de/crp/alpaka-group-container/"
+    container_url += "alpaka-ci-ubuntu" + job[UBUNTU][VERSION]
+
+    # If only the GCC is used, use special gcc version of the container.
+    if job[HOST_COMPILER][NAME] == GCC and job[DEVICE_COMPILER][NAME] == GCC:
+        container_url += "-gcc"
+
+    if (
+        ALPAKA_ACC_GPU_CUDA_ENABLE in job
+        and job[ALPAKA_ACC_GPU_CUDA_ENABLE][VERSION] != OFF
+    ):
+        # Cast cuda version shape. E.g. from 11.0 to 110
+        container_url += "-cuda" + str(
+            int(float(job[ALPAKA_ACC_GPU_CUDA_ENABLE][VERSION]) * 10)
+        )
+        if job[HOST_COMPILER][NAME] == GCC:
+            container_url += "-gcc"
+
+    if (
+        ALPAKA_ACC_GPU_HIP_ENABLE in job
+        and job[ALPAKA_ACC_GPU_HIP_ENABLE][VERSION] != OFF
+    ):
+        container_url += "-rocm" + job[ALPAKA_ACC_GPU_HIP_ENABLE][VERSION]
+
+    # append container tag
+    container_url += ":" + str(container_version)
+    return container_url
+
+
+@typechecked
+def job_variables(job: Dict[str, Tuple[str, str]]) -> Dict[str, str]:
+    """Add variables to the job depending of the job dict.
+
+    Args:
+        job (Dict[str, Tuple[str, str]]): Job dict
+
+    Returns:
+        Dict[str, str]: Dict of {variable name : variable value}.
+    """
+    variables: Dict[str, str] = {}
+
+    return variables
+
+
+@typechecked
+def job_tags(job: Dict[str, Tuple[str, str]]) -> List[str]:
```

Comment:
Could it be that this function is currently not used? How do we like to decide in the future when the generator is creating the jobs which job must be compiled only and where we would like to execute compile only. 
Can this become a new list like the compiler list? With `perform test` and `execute only`?

Reply:
The function is not used at the moment, because no job is auto generated.
If the job is only compiling or also the tests executed is a point, which we need to discuss. There are two options.
1. Create a new parameter field for the job generator. It will be more or less randomly decided, if a job executes the tests or not.
2. Implement a logic in the `job_tags()` function or somewhere else, which decides which job needs to execute the tests.

Reply:
In a offline discussion, we decided to implement a logic, which decides which job is compile only and which is running the tests. I added a comment in the `job_generator.py`, where the logic should be executed.